viskores/exec/SplineEvaluateUniformGrid.h

```diff
@@ -0,0 +1,176 @@
+//============================================================================
+//  The contents of this file are covered by the Viskores license. See
+//  LICENSE.txt for details.
+//
+//  By contributing to this file, all contributors agree to the Developer
+//  Certificate of Origin Version 1.1 (DCO 1.1) as stated in DCO.txt.
+//============================================================================
+
+//============================================================================
+//  Copyright (c) Kitware, Inc.
+//  All rights reserved.
+//  See LICENSE.txt for details.
+//
+//  This software is distributed WITHOUT ANY WARRANTY; without even
+//  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+//  PURPOSE.  See the above copyright notice for more information.
+//============================================================================
+#ifndef viskores_exec_SplineEvaluateUniformGrid_h
+#define viskores_exec_SplineEvaluateUniformGrid_h
+
+#include <viskores/Bounds.h>
+#include <viskores/ErrorCode.h>
+
+namespace viskores
+{
+namespace exec
+{
+
+class VISKORES_ALWAYS_EXPORT SplineEvaluateUniformGrid
+{
+private:
+  using FieldType = viskores::cont::ArrayHandle<viskores::FloatDefault>;
+  using FieldPortalType = typename FieldType::ReadPortalType;
+
+public:
+  VISKORES_CONT SplineEvaluateUniformGrid() = default;
+
+  template <typename ArrayPortalType>
+  VISKORES_CONT SplineEvaluateUniformGrid(const viskores::Vec3f origin,
+                                          const viskores::Vec3f spacing,
+                                          const viskores::Id3 cellDims,
+                                          const ArrayPortalType& field)
+    : Bounds(origin[0],
+             origin[0] + static_cast<viskores::FloatDefault>(cellDims[0] * spacing[0]),
+             origin[1],
+             origin[1] + static_cast<viskores::FloatDefault>(cellDims[1] * spacing[1]),
+             origin[2],
+             origin[2] + static_cast<viskores::FloatDefault>(cellDims[2] * spacing[2]))
+    , Dimensions(cellDims)
+    , Field(field)
+    , Origin(origin)
+    , Spacing(spacing)
```

Comment:
Your `Bounds` computation expects the dimensions of the cells (as is labeled), but the `Dimensions` member needs to have the dimensions of the points, not the cells, to lookup the field values correctly. Fortunately, the caller actually passes the dimensions of the points, which is the lesser of the two evils.

```suggestion
  VISKORES_CONT SplineEvaluateUniformGrid(const viskores::Vec3f origin,
                                          const viskores::Vec3f spacing,
                                          const viskores::Id3 pointDims,
                                          const ArrayPortalType& field)
    : Bounds(origin[0],
             origin[0] + static_cast<viskores::FloatDefault>((pointDims[0] - 1) * spacing[0]),
             origin[1],
             origin[1] + static_cast<viskores::FloatDefault>((pointDims[1] - 1) * spacing[1]),
             origin[2],
             origin[2] + static_cast<viskores::FloatDefault>((pointDims[2] - 1) * spacing[2]))
    , Dimensions(pointDims)
    , Field(field)
    , Origin(origin)
    , Spacing(spacing)
```